### PR TITLE
Add sourcehut engine

### PR DESCRIPTION
**Pull request by v3rmine**
## What does this PR do?

Adds the sourcehut engine (https://sr.ht) using xpath

## Why is this change important?

With some concerns raised about Github, more and more projects are moved to alternatives forges like sourcehut, so it can be a great addition to add them.

## How to test this PR locally?

search `!srht hello world`

### CODE REVIEW

**Comment by return42** [comment]
@joxcat you should rebase .. not merge master .. I would like to repair the commit series and place a update of the languages descriptions on top .. can you please give write permission on this branch to the maintainers / thanks!

In this PR on the right / bottom you find a checkbox ..

image

**Comment by v3rmine** [comment]
@return42 sorry, I'll know for the next time. 
Normally you should already have the permissions to edit the branch, I checked the box while creating the PR :thinking: 
image

File: searx/data/engine_descriptions.json [hunk 1]
```diff
@@ -1229,6 +1308,14 @@
...
    "ref"
+  ],
+  "lib.rs":[
+   "Lib.rs is a catalog of programs and libraries written in the Rust programming language. It includes crates from the crates.io registry, and a few notable projects published only on GitHub or GitLab.",
+   "wikidata"
+  ],
+  "sourcehut":[
+   "software development repository and forge",
+   "wikidata"
   ]
```
Comment (return42) [inline searx/data/engine_descriptions.json:1319]: @joxcat FYI I added a commit on top to get descriptions of lib.rs and sourcehut / this by example is the English description.

> Normally you should already have the permissions to edit the branch, I checked the box while creating the PR

Oh yes, sorry it was my fail :-)

**Review by return42 — APPROVED** [review]
Excellent work thanks for adding one more engine :+1:

**Pull request merged**